Test/SqlDom/TSqlParserTest.cs

```diff
@@ -460,6 +460,28 @@ UPDATE dbo.Departments
                      return procedureStatementBody.StatementList.StartLine.Equals(5) && procedureStatementBody.StatementList.StartColumn.Equals(1);
                  }
             ));
+        }
+
+        [TestMethod]
+        [Priority(0)]
+        [Timeout(GlobalConstants.DefaultTestTimeout)]
+        [SqlStudioTestCategory(Category.UnitTest)]
+        public void ParsingSelectWithNoOptimizerHint()
+        {
+            string input = "";
+            IList<ParseError> errors = null;
+            TSqlScript script = null;
+
+            input = @"Select 1";
+            errors = null;
+            script = (TSqlScript)new TSql80Parser(true).Parse(new StringReader(input), out errors);
+
+            script.Accept(new GenericFragmentVisitor<SelectStatement>(
+                 delegate (SelectStatement select)
+                 {
+                     return select != null && !select.OptimizerHints.Any();
```

Comment:
```suggestion
                     return select != null && !select.OptimizerHints.Any(x => x.FirstTokenIndex == -1 || x.LastTokenIndex == -1);
```

Reply:
well the change I propose doesn't make sense for the test as written.
But the test for #91 should use `select 1 option(recompile)` and be named `ParsingSelectWithOptimizerHint`

Reply:
@vemoo I found the issue and fixed it. Updated the test as well